### PR TITLE
[11.0][IMP] cb_maintenance

### PR DESCRIPTION
**Pull request by jarroyomorales**
Hago PR que asi lo discutimos mejor aqui.
Esta rama contiene el add de maintenance_request_activity y varios improvements al cb_maintenance

### CODE REVIEW

**Comment by jarroyomorales** [comment]
@etobella pensar que hacer con el domain de technician        
 domain="[('id', 'in', maintenance_team_id_member_ids)]",
No se como funciona y que poblemas puede dar asignarle un domain a un campo computado, creo que no tiene sentido. Se deberia aplicar al padre.

File: cb_maintenance/models/maintenance_request.py [hunk 1]
```diff
@@ -21,18 +20,49 @@ class MaintenanceRequest(models.Model):
...
         compute="_compute_maintenance_team_id_member_ids",
     )
     technician_user_id = fields.Many2one(
-        domain="[('id', 'in', maintenance_team_id_member_ids)]", default=False
+        compute="_compute_technician_user_id", store=True
     )
-    manager_id = fields.Many2one("res.users", string="Manager")
+    technician_id = fields.Many2one(
+        "res.partner",
+        # TODO: Add domain
```
Comment (etobella) [inline cb_maintenance/models/maintenance_request.py]: Missing a domain here!

File: cb_maintenance/models/maintenance_request.py [hunk 2]
```diff
@@ -21,18 +20,49 @@ class MaintenanceRequest(models.Model):
...
+        default=False,
+    )
+
+    color = fields.Integer(compute="_compute_color", store=True)
+
+    @api.depends("maintenance_type")
+    def _compute_color(self):
+        for record in self:
+            # Hardcoded colors, green for preventive and red for corrective
+            c_preventive = 10
```
Comment (etobella) [inline cb_maintenance/models/maintenance_request.py]: Maybe on a parameter? Maybe there is other types. What should we do with this? The color will be visible on trees and forms?

File: cb_maintenance/views/maintenance_request.xml [hunk 3]
```diff
@@ -20,20 +20,22 @@
...
             <xpath expr="//span[hasclass('label')]/.." position="after">
-                <field name="solved" invisible="1"/>
                 <div attrs="{'invisible': [('solved', '=', False)]}">
                     <span class="label label-success pull-right">Solved</span>
                 </div>
             </xpath>
 
             <field name="technician_user_id" position="after">
+                <field name="manager_id" options="{'no_open': True, 'no_create': True}"/>
+                <field name="technician_id"/>
```
Comment (etobella) [inline cb_maintenance/views/maintenance_request.xml]: no_open, no_create

File: cb_maintenance/wizards/wizard_create_maintenance_request.py [hunk 4]
```diff
@@ -6,42 +6,47 @@
...
-        })
+        request = self.env["maintenance.request"].create(
+            {
+                "name": self.name,
+                "maintenance_team_id": self.maintenance_team_id.id,
+                "description": self.description or self.name,
+                "location_id": self.location_id.id,
+                "category_id": ctg,
+                "manager_id": self.maintenance_team_id.user_id.id or False,
+                # TODO: Cambiar por un default user de equipo
```
Comment (etobella) [inline cb_maintenance/wizards/wizard_create_maintenance_request.py]: Missing todo

**Review by etobella — CHANGES_REQUESTED** [review]
Some changes. Also, it is solved the part about allowing maintenance users to add directly?

File: cb_maintenance/models/maintenance_request.py (hunk 2 above)
Comment (jarroyomorales) [reply cb_maintenance/models/maintenance_request.py]: I think its not that bad to hardcode this, we can make a decision on it and if in the future we want to change it, we can just do a PR, its not like we want this customizable. Also it is harder to read from the css or other possible places where we may use the values.
At most I would create a global variable at the top of the file so we can import it from other places.

**Comment by jarroyomorales** [comment]
> Some changes. Also, it is solved the part about allowing maintenance users to add directly?

What do you mean with this?

File: cb_maintenance/models/maintenance_request.py [hunk 5]
```diff
@@ -21,18 +22,55 @@ class MaintenanceRequest(models.Model):
...
+    maintenance_color = fields.Char(compute="_compute_color")
+    color = fields.Integer(compute="_compute_color", store=True)
+
+    @api.depends("maintenance_type")
+    def _compute_color(self):
+        for record in self:
+            # Hardcoded colors, green for preventive and red for corrective
+            c_preventive = 10
+            c_corrective = 1
+            if record.maintenance_type == "preventive":
```
Comment (etobella) [inline cb_maintenance/models/maintenance_request.py]: Maybe a hook that returns color and maintenance_color? In order to use properly with works...

File: cb_maintenance/views/maintenance_request.xml [hunk 6]
```diff
@@ -12,28 +12,37 @@
             <form position="attributes">
                 <attribute name="create">false</attribute>
                 <attribute name="delete">false</attribute>
+                <attribute name="edit">false</attribute>
             </form>
 
+            <xpath expr="//header" position="before">
```
Comment (etobella) [inline cb_maintenance/views/maintenance_request.xml]: Still don't like this, but we could talk with them

File: cb_maintenance/views/maintenance_request.xml [hunk 7]
```diff
@@ -90,6 +132,38 @@
...
+    </record>
+
+    <record model="ir.ui.view" id="hr_equipment_request_view_search">
+        <field name="name">maintenance.request.kanban (in cb_maintenance for manager)</field>
+        <field name="model">maintenance.request</field>
+        <field name="inherit_id" ref="maintenance.hr_equipment_request_view_search"/>
+        <field name="arch" type="xml">
+            <filter name="my_requests" position="attributes">
+                <attribute name="domain">['|', '|', ('technician_user_id', '=', uid), ('manager_id', '=', uid), ('owner_user_id', '=', uid)]</attribute>
+            </filter>
```
Comment (etobella) [inline cb_maintenance/views/maintenance_request.xml:162]: Create two more
 - "On my hands": [('follower_id', '=', uid)]
 - "Managed by me": [('manager_id', '=', uid)]
also, this fields should be searchable

**Review by etobella — CHANGES_REQUESTED** [review]

**Pull request merged**